deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,262 @@
+
+    In this class, we establish a sequential model for the Deep Tensor Neural Network (DTNN) [1]_.
+
+    References
+    ----------
+    .. [1] Schütt, Kristof T., et al. "Quantum-chemical insights from deep
+        tensor neural networks." Nature communications 8.1 (2017): 1-8.
+
+    """
+
+    def __init__(self,
+                 n_tasks,
+                 n_embedding=30,
+                 n_hidden=100,
+                 n_distance=100,
+                 distance_min=-1,
+                 distance_max=18,
+                 output_activation=True,
+                 mode="regression",
+                 dropout=0.0,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        n_tasks: int
+            Number of tasks
+        n_embedding: int, optional
+            Number of features per atom.
+        n_hidden: int, optional
+            Number of features for each molecule after DTNNStep
+        n_distance: int, optional
+            granularity of distance matrix
+            step size will be (distance_max-distance_min)/n_distance
+        distance_min: float, optional
+            minimum distance of atom pairs, default = -1 Angstorm
+        distance_max: float, optional
+            maximum distance of atom pairs, default = 18 Angstorm
+        mode: str
+            Only "regression" is currently supported.
+        dropout: float
+            the dropout probablity to use.
+
+        """
+        super(DTNN, self).__init__()
+        if mode not in ['regression']:
+            raise ValueError("Only 'regression' mode is currently supported")
+        self.n_tasks = n_tasks
+        self.n_embedding = n_embedding
+        self.n_hidden = n_hidden
+        self.n_distance = n_distance
+        self.distance_min = distance_min
+        self.distance_max = distance_max
+        self.step_size = (distance_max - distance_min) / n_distance
+        self.steps = np.array(
+            [distance_min + i * self.step_size for i in range(n_distance)])
+        self.steps = np.expand_dims(self.steps, 0)
+        self.output_activation = output_activation
+        self.mode = mode
+        self.dropout = dropout
+
+        # get DTNNEmbedding
+        self.dtnn_embedding = layers.DTNNEmbedding(n_embedding=self.n_embedding)
+
+        # get DTNNSteps
+        self.dtnn_step = layers.DTNNStep(n_embedding=self.n_embedding,
+                                         n_distance=self.n_distance)
+
+        # get DTNNGather
+        self.dtnn_gather = layers.DTNNGather(
+            n_embedding=self.n_embedding,
+            layer_sizes=[self.n_hidden],
+            n_outputs=self.n_tasks,
+            output_activation=self.output_activation)
+
+        # get Final Linear Layer
+        self.linear = nn.LazyLinear(self.n_tasks)
+
+    def forward(self, inputs: List[torch.Tensor]):
+        """
+        Parameters
+        ----------
+        inputs: List
+            List of Tensors containing.
```

Comment:
Should the list be in this order always?

Reply:
Yes, they should always be in the same order.

Reply:
Can you then mention that point there, it can be a little vague for some

Reply:
Can you mention that in the comments, it's can be bit vague for some

Reply:
Done